nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -1,213 +1,907 @@
 # Xen hypervisor (Dom0) support.
 
-{ config, lib, pkgs, ... }:
-
-with lib;
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
 
 let
   cfg = config.virtualisation.xen;
+
+  xenBootBuilderPath = "${
+    pkgs.writeShellApplication {
+      name = "xenBootBuilder";
+      runtimeInputs =
+        (with pkgs; [
+          binutils
+          coreutils
+          findutils
+          gawk
+          gnugrep
+          gnused
+          jq
+        ])
+        ++ lib.lists.optionals (cfg.efi.bootBuilderFlag == "--report") (
+          with pkgs;
+          [
+            bat
+            diffutils
+          ]
+        );
+
+      runtimeEnv = {
+        efiMountPoint = config.boot.loader.efi.efiSysMountPoint;
+      };
+      text =
```

Comment:
Could this be a separate `.bash` file that is referenced from the Nix file? I think it would be more readable. The Nix comments could just be Bash comments; I don’t think avoiding rebuilds for that is worth enough to keep this as one huge string literal.

Reply:
The script is now stored in a separate file.